src/main/kotlin/com/infusionvlc/somniumserver/dreams/usecases/SearchDream.kt

```diff
@@ -0,0 +1,17 @@
+package com.infusionvlc.somniumserver.dreams.usecases
+
+import com.infusionvlc.somniumserver.dreams.models.Dream
+import com.infusionvlc.somniumserver.dreams.persistence.DreamRepository
+import com.infusionvlc.somniumserver.dreams.persistence.toDomain
+import org.springframework.data.domain.PageRequest
+import org.springframework.stereotype.Component
+
+@Component
+class SearchDream(
+  private val dao: DreamRepository
+) {
+  operator fun invoke(title: String, page: Int, pageSize: Int): List<Dream> =
```

Comment:
I like the `invoke` trick but we are using `execute` for not confusing people for now.

I have not written a `UseCase` interface because each one has its own parameters and results types and every time I create this kind of interfaces I end up with two or three files like `UseCase<P, R>`, `NoParamUseCase<R>`, `NoResultUseCase<P>`.

Reply:
That usecase needs a refactor, I totally agree. The thing is I'm still thinking how am I going to do it. I can use bindings (which can be even more confusing), I can use function composition and use arrow utility functions `andThen` or `compose` or I can use private extension functions like you proposed 🤔